### DISCUSSION TITLE
Controlling the sequence/order of a cut from SVG

### DISCUSSION THREAD

**Discussion by AnotherRedcoat** [arkypita/LaserGRBL#2282] [Q&A]
Is it possible to force LaserGRBL to generate cuts/engraves in a specific order?

I want the smaller inner pieces cut first to reduce the risk of air-assist blowing the holding piece away, and the outer bounding box cut last.

In Inkscape, I can, using layers and the sequence of paths within them, force the SVG to contain the elements in the right order (basically outputs them in bottom-up order), which LaserGRBL more or less follows, except it's always doing the bounding box first.

If someone could explain the current logic, perhaps I can build the SVG around that. Alternatively, if LaserGRBL had a feature where it optionally cuts from inner to outer vs whatever the default currently is, that would also work.

![Kurf_Calibration_BG](https://github.com/arkypita/LaserGRBL/assets/117658333/e8f951ab-21e6-458c-9ebf-d2f5474e940e)

In the example attached SVG file, an v7.6.0 exported NC file will cut rect3/OuterBox first, then path2/AlmostACircle, then rect2/SquareKeyhole then path1/Squiggle. The expected SVG order would be mostly the same order except OuterBox comes last.

**Comment by Tribo-2dot0** [comment]
Em Inkscape se você colocar a Caxa Delimitadora (contorno) na camada superior ele deveria cortar por último; a ordem dos objetos dentro de uma camada também tem o mesmo efeito, mas prevalece a ordem das camadas...
